### ISSUE TITLE
Make `--without-view` the default for environments

### ISSUE THREAD

**Issue by kwryankrattiger** [LLNL/spack#47198]
### Summary

Environments should be created without a view by default.

### Rationale

Views have multiple implications that can cause issues in user environments and workflows. This is due to the inherent property of views that modifies the shell environment to point at Spack installed packages in the `PATH`. This is problematic as it can break other applications that search paths from variables and suddenly start finding incompatible packages.

An example of this is trying to open files with Vim when I have a spack environment activate with a view that installs libraries that Vim uses, but are the wrong version or are missing features. This makes it impossible to work from a shell with that environment activated.

Views are still useful tools for exposing Spack environments in a `venv` sort of way, and should be an option for users who expect to use it in that way. But they have too many additional implications associated with them to make sense as a sane default.

### Description

Changing the default for environments to set the view to `False`.

### Additional information

CC: @haampie 

### General information

- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
> inherent property of views that modifies the shell environment to point at Spack installed packages in the PATH

If I understood Ryan's issue correctly, it was similar to how LD_LIBRARY_PATH broke system executables, except this time the culprit was PYTHONPATH. An editor would link to system libpython.so, whose plugins would then search the environment view's lib directory due to PYTHONPATH, and find incompatible / conflicting packages.

So, the issue is not necessarily with the view per se, but rather with shell support of the view.

**Comment by kwryankrattiger** [comment]
Correct. My number one gripe with views is they modify the environment in ways that can break tools on the system. Python is a prime example of that.

I think they also kind of need to to get things setup correctly so that using the tools installed in the view work right? And I don't want to break views for people who want to use them that way, I just don't want to pollute my shell just because I am in an environment with a view.

I would also be happy to have a separate command to activate the environments view.

Something like

```console
spack [-e env_name] view activate
spack env activate --view env_name
```

**Comment by kwryankrattiger** [comment]
I will also say here, my other gripe is views take, potentially, a long to to rebuild themselves after an install. So even if the environment pollution is mitigated with another command there is still the added overhead of creating the view by default.

Most of the time I am not using the view, nor deploying with a view. I generate modules because that is what the HPC world uses by default. So the view is still sitting as unnecessary overhead to one of the major supported workflows.

**Comment by haampie** [comment]
Here's another argument, if people do:

```
spack env create x
spack env activate x
spack install --add y
```

then the next time they run

```
spack install --add z
```

the build environment is potentially polluted by the view in PATH/PKG_CONFIG_PATH/etc.
